hummingbot/strategy/pure_market_making/pure_market_making_v2.pyx

```diff
@@ -356,6 +359,121 @@ cdef class PureMarketMakingStrategyV2(StrategyBase):
+                    from_currency=flat_fee_currency,
+                    to_currency=quote_asset
+                )
+        return total_flat_fees
+
+    cdef object c_add_transaction_costs_to_pricing_proposal(self,
+                                                            object market_info,
+                                                            object pricing_proposal,
+                                                            object sizing_proposal):
+        cdef:
+            MarketBase maker_market = market_info.market
+            OrderBook maker_orderbook = market_info.order_book
+            object buy_prices_with_tx_costs = pricing_proposal.buy_order_prices
+            object sell_prices_with_tx_costs = pricing_proposal.sell_order_prices
+            int64_t current_tick = <int64_t>(self._current_timestamp // self._status_report_interval)
+            int64_t last_tick = <int64_t>(self._last_timestamp // self._status_report_interval)
+            bint should_report_warnings = ((current_tick > last_tick) and
+                                           (self._logging_options & self.OPTION_LOG_STATUS_REPORT))
+            # Current warning report threshold is 10%
+            # If the adjusted price with transaction cost is 10% away from the suggested price,
+            # warnings will be displayed
+            double warning_report_threshold = 0.1
+
+        # If both buy order and sell order sizes are zero, no need to add transaction costs
+        # as no new orders are created
+        if sizing_proposal.buy_order_sizes[0] == s_decimal_zero and \
+                sizing_proposal.sell_order_sizes[0] == s_decimal_zero:
+            return pricing_proposal
+
+        buy_index = 0
+        for buy_price, buy_amount in zip(pricing_proposal.buy_order_prices,
+                                         sizing_proposal.buy_order_sizes):
+            if buy_amount > s_decimal_zero:
+                fee_object = maker_market.c_get_fee(
+                    market_info.base_asset,
+                    market_info.quote_asset,
+                    OrderType.LIMIT,
+                    TradeType.BUY,
+                    buy_amount,
+                    buy_price
+                )
+                # Total flat fees charged by the exchange
+                total_flat_fees = self.c_sum_flat_fees(market_info.quote_asset,
+                                                       fee_object.flat_fees)
+                # Find the fixed cost per unit size for the total amount
+                # Fees is in Float
+                fixed_cost_per_unit = total_flat_fees / float(buy_amount)
+                # New Price = Price * (1 - maker_fees) - Fixed_fees_per_unit
+                buy_price_with_tx_cost = float(buy_price) * (1 - fee_object.percent) - fixed_cost_per_unit
+            else:
+                buy_price_with_tx_cost = buy_price
+
+            buy_price_with_tx_cost = maker_market.c_quantize_order_price(market_info.trading_pair,
```

Comment:
There's actually a second risk with the equation here, besides just a large correction...

If `buy_amount` is very small.. then it's possible for `fixed_cost_per_unit` to exceed `buy_price` to begin with and arrive at a zero or negative price.

So we're gonna need another check for 0 or negative prices as well.